### DISCUSSION TITLE
HxInputDate FocusAsync() Not Working

### DISCUSSION THREAD

**Discussion by chobo2** [havit/Havit.Blazor#785] [Q&A]
Hi

I am trying to use "FocusAsync" but it never focuses on it.

I tried taking the default code and added a reference

```
<HxInputDate Label="Date" @bind-Value="@value" @ref="testInputDate" />

@code {
    public DateTime value { get; set; }
   private HxInputDate<DateTime> testInputDate;
   protected override async Task OnAfterRenderAsync(bool firstRender)
    {
        if (firstRender)
        {
              testInputDate().FocusAsync()
        }
    }
}
```

In my actual application I get an error telling me that it can't be done after first render or soemthing like that though it could be because the HxInputDate is in a child component but since I can't even get it working with the example code I thought I would start there.

Also on a side note, is there away to get the calendar to put when you tab into a HxInputDate?

**Comment by chobo2** [comment]
anyone know if this is a bug or just how the date pickers work?

**Comment by hakenr** [comment]
This was a bug #791, which should be fixed now.

**Comment by hakenr** [comment]
@chobo2 Sorry, I forgot to link this discussion to the bug report to let you know it was confirmed and fixed.

**Reply by chobo2** [reply]
Thanks, I added by findings to the bug ticket.

**Reply by hakenr** [reply]
@chobo2 Thanks for the new findings, I adjusted the inputs `FocusAsync()` method to open the dropdown.
